internal/transport/http2_server.go

```diff
@@ -1199,12 +1189,15 @@ func (t *http2Server) keepalive() {
 // Close starts shutting down the http2Server transport.
 // TODO(zhaoq): Now the destruction is not blocked on any pending streams. This
 // could cause some resource issue. Revisit this later.
-func (t *http2Server) Close() {
+func (t *http2Server) Close(err error) {
 	t.mu.Lock()
 	if t.state == closing {
 		t.mu.Unlock()
 		return
 	}
+	if logger.V(logLevel) {
+		logger.Infof("Closing transport, will close the connection. Err: %v", err)
```

Comment:
Same comment here about the log message being too verbose. Concise messages (without compromising on the information to be conveyed) will help greatly during debugging. So, please do consider how these logs will show up. Run a test or two and see how these show up.